contentcuration/contentcuration/static/js/edit_channel/channel_set/views.js

```diff
@@ -0,0 +1,129 @@
+import Vue from 'vue';
+import ChannelSetModalComponent from './views/ChannelSetModal.vue';
+
+var Backbone = require('backbone');
+var BaseViews = require("../views");
+var dialog = require("../utils/dialog");
+var store = require('./vuex/store');
+var vueIntl = require("vue-intl");
+var translations = require("utils/translations");
+var { PageTypes } = require('./constants');
+
+var ChannelSetModal = Vue.extend(ChannelSetModalComponent);
+
+// Flatten translation dictionary
+var unnested_translations = {};
+Object.keys(translations).forEach(function (key) {
+    Object.keys(translations[key]).forEach(function(nestedKey) {
+        unnested_translations[key + "." + nestedKey] = translations[key][nestedKey];
+    });
+});
+
+Vue.use(vueIntl, {"defaultLocale": "en"});
+
+var currentLanguage = "en";
+if (global.languageCode) {
+    currentLanguage = global.languageCode;
+    Vue.setLocale(currentLanguage);
+}
+
+Vue.registerMessages(currentLanguage, unnested_translations);
+Vue.prototype.$tr = function $tr(messageId, args) {
+    const nameSpace = this.$options.name;
+    if (args) {
+        if (!Array.isArray(args) && typeof args !== 'object') {
+            logging.error(`The $tr functions take either an array of positional
+                            arguments or an object of named options.`);
+        }
+    }
+    const defaultMessageText = this.$options.$trs[messageId];
+    const message = {
+        id: `${nameSpace}.${messageId}`,
+        defaultMessage: defaultMessageText,
+    };
+
+    return this.$formatMessage(message, args);
+};
+
+
+var NAMESPACE = "channel_set";
+var MESSAGES = {
+    "warning_message": "Any associated content will not be imported or referenced as related content.",
+    "importing_content": "Importing Content...",
+    "save_and_close": "SAVE & CLOSE",
+    "dont_save": "DON'T SAVE",
+    "keep_open": "KEEP OPEN"
+}
+
+function checkForSave(state) {
+  return state.channel_set.saving;
+}
+
+var ChannelSetModalView = BaseViews.BaseView.extend({
+    name: NAMESPACE,
+    $trs: MESSAGES,
+    initialize: function(options) {
+        this.options = options;
+        this.statusWatcher = store.watch(
+          checkForSave,
+          this._handleSaving.bind(this)
+        );
+        this.render();
+    },
+
+    render: function() {
+        Vue.nextTick().then(this._mountVueComponent.bind(this));
```

Comment:
@DXCanas is there anything else that you came across in your preview renderer work that might help prevent Backbone/Vue related issues here? Do we now have a standard enough format to be able to make a BaseVueView that we can use for this kind of thing generally?

Reply:
The approach here is pretty fundamentally different from mine.

As much as I could, I limited Backbone -> Vue communication to "Here's some transformed data for your props"
and Vue -> Backbone communication to "Here's an HTML element. Put it where you please"

Because of this, my "glue" solution was [pretty slim](https://github.com/learningequality/studio/pull/752/files#diff-5f6672f0523a6ea60b2d04bd7d94859b).

This seems more complex, mostly out of necessity. One thing I'm having trouble understanding is why we chose to have this `Backbone.View` interact with the store rather than having vue components dispatch the actions directly? Haven't been able to take a close look, but it seems like the components are already committing to the store anyway? 

However, I think we could simplify some of this logic by leveraging [`v-if` behavior](https://vuejs.org/v2/api/#vm-destroy) to handle `$destroy`. I handled that by hooking into a `Backbone.Event`, but using a getter for that purpose would probably clean some of this up.

We could also clean up some of the `_resetPageState` logic by hooking into `$beforeDestroy` hooks?

Reply:
I think we can punt this to "let's work out a consistent way to do this" for now, but keep the conversation ongoing.

Reply:
Started a page here: https://www.notion.so/learningequality/Establish-best-practices-3443b679851b4645a985aad77eabfbd5